mountpoint-s3/src/prefetch.rs

```diff
@@ -115,27 +113,27 @@ pub struct MetadataRanges {
     pub rowgroup_col_ranges: RowgroupColRanges,
 }
 
-pub type MetadataRef = Arc<AsyncRwLock<Option<MetadataRanges>>>;
-pub type RawMetadataRef = Arc<AsyncRwLock<Option<RawMetadata>>>;
-
 #[derive(Debug, Clone, Default)]
 pub struct CacheEntry {
-    pub raw_metadata: RawMetadataRef,
-    pub parsed_metadata: MetadataRef,
-    pub in_memory_cache: InMemoryCacheRef,
+    pub state: Arc<AsyncRwLock<Option<CacheEntryState>>>,
 }
-type CachedRef = Arc<DashMap<ObjectId, CacheEntry>>;
+
+#[derive(Debug)]
+pub struct CacheEntryState {
+    raw_metadata: RawMetadata,
+    parsed_metadata: MetadataRanges,
+    in_memory_cache: InMemoryRecord,
+}
+
+type MetadataCache = DashMap<ObjectId, CacheEntry>;
```

Comment:
You should be able to move all these types to the `parquet_stream` module and let `ParquetPartStream::new` create the cache.